### PR TITLE
[SPARK-13474][PROJECT INFRA] Update packaging scripts to push artifacts to home.apache.org

### PR DESCRIPTION
**Pull request by JoshRosen**
Due to the people.apache.org -> home.apache.org migration, we need to update our packaging scripts to publish artifacts to the new server. Because the new server only supports sftp instead of ssh, we need to update the scripts to use lftp instead of ssh + rsync.

### CODE REVIEW

**Comment by JoshRosen** [comment]
Argh, looks like I need to fix some LFTP problems. Hold off on merge or review for now.

**Comment by srowen** [comment]
+1

File: dev/create-release/release-build.sh [hunk 1]
```diff
@@ -105,10 +121,10 @@ rm -rf .git
 cd ..
 
 if [ -n "$REMOTE_PARENT_MAX_LENGTH" ]; then
-  old_dirs=$($SSH $USER_HOST ls -t $REMOTE_PARENT_DIR | tail -n +$REMOTE_PARENT_MAX_LENGTH)
+  old_dirs=$(LFTP nlist $REMOTE_PARENT_DIR | sort | head -n +$REMOTE_PARENT_MAX_LENGTH)
```
Comment (JoshRosen) [inline dev/create-release/release-build.sh]: It turns out that `nlist` will return `./` and `../`, so we need to filter those out here.

File: dev/create-release/release-build.sh [hunk 2]
```diff
@@ -178,11 +198,11 @@ if [[ "$1" == "package" ]]; then
...
   echo "Copying release tarballs to $dest_dir"
-  $SSH $USER_HOST mkdir $dest_dir
-  rsync -e "$SSH" spark-* $USER_HOST:$dest_dir
+  LFTP mkdir -p $dest_dir
+  LFTP mput -O $dest_dir spark-*
   echo "Linking /latest to $dest_dir"
-  $SSH $USER_HOST rm -f "$REMOTE_PARENT_DIR/latest"
-  $SSH $USER_HOST ln -s $dest_dir "$REMOTE_PARENT_DIR/latest"
+  LFTP rm -f "$REMOTE_PARENT_DIR/latest"
+  LFTP ln -s $dest_dir "$REMOTE_PARENT_DIR/latest"
```
Comment (JoshRosen) [inline dev/create-release/release-build.sh]: The FTP protocol itself does not support symlinks and the older versions of `lftp` don't have the `ln` command. I'm going to work around this by uploading the binaries twice, once to the `latest` folder and once to the other one. With the symlink, at least the change to `/latest` appeared relatively atomic so you wouldn't see half-uploaded files in there. I'll try to work around this by uploading to dest_dir, moving, then uploading again. This isn't perfect but seems like a decent hack.

File: dev/create-release/release-build.sh [hunk 3]
```diff
@@ -178,11 +198,17 @@ if [[ "$1" == "package" ]]; then
...
+  set -x
+  # Put to new directory:
+  LFTP mkdir -p $dest_dir
+  LFTP mput -O $dest_dir spark-*
+  # Delete /latest directory and rename new upload to /latest
+  LFTP rm -rf "$REMOTE_PARENT_DIR/latest"
+  LFTP mv $dest_dir "$REMOTE_PARENT_DIR/latest"
+  # Re-upload a second time and leave the files in the timestamped upload directory:
+  LFTP mkdir -p $dest_dir
+  LFTP mput -O $dest_dir spark-*
```
Comment (JoshRosen) [inline dev/create-release/release-build.sh]: I might need to quote the `spark-*` so that the parameter expansion is done by `lftp` rather than the shell.

**Comment by JoshRosen** [comment]
This is now working for package snapshots; will test for doc snapshots soon. See https://home.apache.org/~pwendell/spark-nightly/spark-master-bin/ for what the results look like.

**Comment by JoshRosen** [comment]
Woohoo! Doc publishing seems to be working now (albeit slowly), so this should be ready for a final sign-off and merge.

**Comment by JoshRosen** [comment]
Ping @marmbrus for sign-off, since this impacts your 1.6.1 RC. I'll backport this change to branch-1.6 so you can cut that branch.

**Comment by marmbrus** [comment]
LGTM.  Thanks for doing this!

**Comment by JoshRosen** [comment]
I'm going to merge this into master and branch-1.6 so that I can begin packaging a 1.6.1 release candidate.

**Pull request closed without merging**